cockatrice/src/zoneviewwidget.cpp

```diff
@@ -185,6 +185,24 @@ void ZoneViewWidget::paint(QPainter *painter, const QStyleOptionGraphicsItem *op
 
 void ZoneViewWidget::moveWidget(QPointF scenePos)
 {
+    if(scenePos.x() < 0)
+    {
+        scenePos.setX(0);
+    } else {
+        qreal maxw = scene()->sceneRect().width() - 100;
+        if(scenePos.x() > maxw)
+            scenePos.setX(maxw);
+    }
```

Comment:
Sounds like less lines of code, but more cycles of cpu:
 * if the value is already ok (99% of the time), you'll just call setX() with the very same value;
 * if the value is less than 0,  std::max() will set it to 0; in this case we already know that 0 is less than maxw, so there's no need to std::min();
 * the value of maxw is always calculated, even if it's not really needed;

I agree that your code looks nicer, but given that this method is called on a mousemove event that can be triggered tens of times per second, i'd prefer to keep it faster.